core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -396,8 +400,22 @@ default Iterator<RelationCommentMetadata> streamRelationComments(ConnectorSessio
                         return RelationCommentMetadata.forRelation(tableName, getTableMetadata(session, tableHandle).getComment());
                     }
                     catch (RuntimeException e) {
-                        // getTableHandle or getTableMetadata failed call may fail if table disappeared during listing or is unsupported.
-                        Helper.juliLogger.log(Level.WARNING, e, () -> "Failed to get metadata for table: " + tableName);
+                        boolean silent = false;
+                        if (e instanceof TrinoException trinoException) {
+                            ErrorCode errorCode = trinoException.getErrorCode();
+                            silent = errorCode.equals(UNSUPPORTED_TABLE_TYPE.toErrorCode()) ||
```

Comment:
I think it would be nice to define this condition once and reuse it both places.

Reply:
Not possible without exposing this as a SPI functionality, which i wanted to avoid.

Ultimately each connector will implement this method directly, so the logic will become connector-specific anyway